src/main/kotlin/com/lovelysystems/gradle/LSAws.kt

```diff
@@ -1,15 +1,28 @@
 package com.lovelysystems.gradle
 
+import com.lovelysystems.gradle.aws.S3DownloadTaskConfiguration
+import com.lovelysystems.gradle.aws.S3UploadDirectoryTaskConfiguration
+import com.lovelysystems.gradle.aws.downloadS3File
+import com.lovelysystems.gradle.aws.uploadS3Directory
 import org.gradle.api.Project
 import org.gradle.kotlin.dsl.getValue
 import org.gradle.kotlin.dsl.invoke
 import org.gradle.kotlin.dsl.provideDelegate
 import org.gradle.kotlin.dsl.registering
+import software.amazon.awssdk.regions.Region
 import java.io.ByteArrayOutputStream
 
 private const val AWS_GROUP = "aws"
 
-fun Project.awsProject(profile: String) {
+class AwsProjectConfiguration {
+    var downloadTasks: List<S3DownloadTaskConfiguration.() -> Unit> = listOf()
```

Comment:
this is actually an antipattern to receive lists of tasks, we should not rely on task configs in that early phase of the build. 

 config should be like:

```
lovely {
   awsProjedct(justsettings...)
}

// task config
val upload1 by tasks.registering<Theuploadclass>() {
}
....

see also https://docs.gradle.org/current/userguide/lazy_configuration.html


```